src/pal/src/map/map.cpp

```diff
@@ -2241,7 +2241,9 @@ MAPmmapAndRecord(
     PAL_ERROR palError = NO_ERROR;
     LPVOID pvBaseAddress = NULL;
 
-    pvBaseAddress = mmap(addr, len, prot, flags, fd, offset);
+    off_t adjust = offset & (VIRTUAL_PAGE_MASK);
+
+    pvBaseAddress = mmap(static_cast<char *>(addr) - adjust, len + adjust, prot, flags, fd, offset - adjust);
```

Comment:
Also, on amd64 where we use the executable allocator to ensure that the virtual address space where we map the image is located close to the libcoreclr.so, this change breaks that functionality. When a section cannot be mapped to the requested address because the previous section mappings have consumed more virtual address space than expected, mmap can map it to a random address in the whole VA space. That would effectively kill the performance benefits we get from mapping close to the libcoreclr.so.
To fix that, it seems one way would be to augment the sectionBase in the MAPMapPEFile for each call to MAPmmapAndRecord to reflect the expected mapping address based on the page size.